llm4pddl/envs/assets/pddl/dressed/generate_tasks.py

```diff
@@ -0,0 +1,199 @@
+    """
+    problems = []
+    for _ in range(1, num_probs + 1):
+        num_people = rng.integers(min_people, max_people)
+        num_casual = rng.integers(min_casual, max_casual)
+        num_formal_dress = rng.integers(min_formal_dress, max_formal_dress)
+        num_formal_suit = rng.integers(min_formal_suit, max_formal_suit)
+        num_extra = rng.integers(min_extra_clothes, max_extra_clothes)
+        if num_people < num_casual + num_formal_dress + num_formal_suit:
+            num_people = num_casual + num_formal_dress + num_formal_suit
+
+        problem = _generate_dressing_problem(num_people, num_casual,
+                                             num_formal_dress, num_formal_suit,
+                                             num_extra, rng)
+        problems.append(problem)
+    return problems
+
+
+def _generate_dressing_problem(num_people: int, num_casual_events: int,
+                               num_formal_in_dress: int,
+                               num_formal_in_suit: int, num_extra_objects: int,
+                               rng: np.random.Generator) -> str:
+    """Generates a single problem with the given parmeters."""
+    assert num_people >= (num_casual_events + num_formal_in_dress +
+                          num_formal_in_suit)
+    # deciding extra clothing:
+    extras = ['ex_p', 'ex_d', 'ex_sp', 'ex_ss', 'ex_np', 'ex_cs', 'ex_sj']
+    exs = {ex: 0 for ex in extras}
+    for _ in range(num_extra_objects):
+        exs[rng.choice(extras)] += 1
+    # making objects:
+    persons = [f'person{i}' for i in range(1, num_people + 1 + exs['ex_p'])]
+    dresses = [
+        f'dress{i}' for i in range(1, num_formal_in_dress + 1 + exs['ex_d'])
+    ]
+    sweatpants = [
+        f'sweatpants{i}'
+        for i in range(1, num_casual_events + 1 + exs['ex_sp'])
+    ]
+    sweatshirts = [
+        f'sweatshirt{i}'
+        for i in range(1, num_casual_events + 1 + exs['ex_ss'])
+    ]
+    nicepants = [
+        f'nice-pants{i}'
+        for i in range(1, num_formal_in_suit + 1 + exs['ex_np'])
+    ]
+    collaredshirt = [
+        f'collared-shirt{i}'
+        for i in range(1, num_formal_in_suit + 1 + exs['ex_cs'])
+    ]
+    suitjacket = [
+        f'suit-jacket{i}'
+        for i in range(1, num_formal_in_suit + 1 + exs['ex_d'])
+    ]
+    objects = [{
+        'type': 'person',
+        'objs': persons
+    }, {
+        'type': 'dress',
+        'objs': dresses
+    }, {
+        'type': 'sweatpants',
+        'objs': sweatpants
+    }, {
+        'type': 'sweatshirt',
+        'objs': sweatshirts
+    }, {
+        'type': 'nice-pants',
+        'objs': nicepants
+    }, {
+        'type': 'collared-shirt',
+        'objs': collaredshirt
+    }, {
+        'type': 'suit-jacket',
+        'objs': suitjacket
+    }]
+    object_str = ''
+    for obj in objects:
+        if len(obj['objs']) > 0:
+            object_str += ' '.join(
+                obj['objs']) + f' - {obj["type"]}\n            '
```

Comment:
minor: I don't really care about standardizing double quotes vs single quotes, but since we have both here in the same line, it looks a little funny -- change one or the other to match

Reply:
the reason i used the difference here is that I am using quotes within the f string so I cant use `'` throughout. would it be better if i did` \'` instead of `"` for the inner string?

Reply:
Ohh, I just missed that. Fine to leave as is!